src/rate-limiter.ts

```diff
@@ -20,7 +20,12 @@ export class RateLimiter {
 
     public checkForLimiting = (xmlHttpRequest: XMLHttpRequest): void => {
         try {
-            const response: CaptureResponse = JSON.parse(xmlHttpRequest.responseText)
+            const text = xmlHttpRequest.responseText
+            if (!text || !text.length) {
+                return
+            }
+
+            const response: CaptureResponse = JSON.parse(text)
```

Comment:
this is the actual fix (ignoring the noise from refactoring logger into its own file to make it easier to mock the smallest area in the test)